### PR TITLE
LOG4J2-1748 and LOG4J2-1780

### PR DESCRIPTION
**Pull request by mikaelstaldal**
Remove executor service from LoggerContext.

### CODE REVIEW

File: log4j-core/src/main/java/org/apache/logging/log4j/core/appender/mom/kafka/KafkaManager.java [hunk 1]
```diff
@@ -73,17 +74,19 @@ public boolean releaseSub(final long timeout, final TimeUnit timeUnit) {
...
                 }
-            };
+            }, "KafkaManager-CloseThread");
+            closeThread.setDaemon(true); // avoid blocking JVM shutdown
+            closeThread.start();
             try {
-                getLoggerContext().submitDaemon(task).get(timeout, timeUnit);
-            } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                closeThread.join(timeUnit.toMillis(timeout));
+            } catch (final InterruptedException ignore) {
```
Comment (jvz) [inline log4j-core/src/main/java/org/apache/logging/log4j/core/appender/mom/kafka/KafkaManager.java]: Shouldn't you reset the interrupt flag after catching this?

**Comment by mikaelstaldal** [comment]
I like my code :-)

(testing reviewing)

File: log4j-core/src/main/java/org/apache/logging/log4j/core/appender/mom/kafka/KafkaManager.java (hunk 1 above)
Comment (mikaelstaldal) [inline log4j-core/src/main/java/org/apache/logging/log4j/core/appender/mom/kafka/KafkaManager.java]: Should I? Why?

Comment (jvz) [inline log4j-core/src/main/java/org/apache/logging/log4j/core/appender/mom/kafka/KafkaManager.java]: The thread would have been interrupted for a reason, so it's generally good to handle the exception and reset the interrupt flag (it doesn't necessarily cause an exception to be thrown; code has to manually check a thread's interrupt status according to the javadocs).

Comment (mikaelstaldal) [inline log4j-core/src/main/java/org/apache/logging/log4j/core/appender/mom/kafka/KafkaManager.java]: OK.

**Comment by jvz** [comment]
Did GitHub parse your comment or did you manually approve the PR?

**Comment by rgoers** [comment]
I manually approved it.

**Pull request merged**